planck-c/edn.c

```diff
@@ -471,61 +470,3 @@ int clj_read_error(char *str, const clj_Reader *r, clj_Result result) {
   return sprintf(str, "%s at line %d, column %d",
                  result_message(result), r->line, r->column);
 }
-
-
-// Print forms
-
-static void print_string(clj_Printer *p, const wchar_t *s) {
-  const wchar_t *i;
-  for (i = s; *i != L'\0'; i++) {
-    p->putwchar(*i);
-  };
-}
-
-#pragma GCC diagnostic ignored "-Wall"
-
-void clj_print(clj_Printer *p, const clj_Node *node) {
-  switch (node->type) {
-
-    case CLJ_NUMBER:
-    case CLJ_STRING:
-    case CLJ_REGEX:
-    case CLJ_SYMBOL:
-    case CLJ_KEYWORD:
-    case CLJ_CHARACTER:
-      print_string(p, node->value);
-      break;
-
-    case CLJ_LIST:
-      p->putwchar(L'(');
-      break;
-    case CLJ_LIST | CLJ_END:
-      p->putwchar(L')');
-      break;
-
-    case CLJ_VECTOR:
-      p->putwchar(L'[');
-      break;
-    case CLJ_VECTOR | CLJ_END:
-      p->putwchar(L']');
-      break;
-
-    case CLJ_MAP:
-      p->putwchar(L'{');
-      break;
-    case CLJ_MAP | CLJ_END:
-      p->putwchar(L'}');
-      break;
-
-    case CLJ_SET:
-      print_string(p, L"#{");
-      break;
-    case CLJ_SET | CLJ_END:
-      p->putwchar(L'}');
-      break;
-
-    default:
-      fatal("unexpected node type");
-  }
-  p->putwchar(L'\n');
-}
```

Comment:
Sorry for not being able to get this on my own but why is the above no long necessary?

Reply:
This code causes warnings because it employs the or operator `|` to form switch cases and the compiler sees that these values are not in the enum. It turns out this code is not used at all. Planck uses this translation unit (originally written by Brandon Bloom and copied into Planck's source tree), to read EDN configuration, but it never uses it to write out EDN. So the simplest solution is to delete it. (Also, since the code is unused, the compiler might warn about that fact.)